src/common/sendpacket.c

```diff
@@ -444,7 +470,18 @@ sendpacket(sendpacket_t *sp, const u_char *data, size_t len, struct pcap_pkthdr
             }
 #endif /* HAVE_NETMAP */
             break;
-
+        case SP_TYPE_LIBXDP:
+            #ifdef HAVE_LIBXDP
+            retcode = len;
+            xsk_ring_prod__submit(&(sp->xsk_info->tx), sp->pckt_count); //submit all packets at once
+            sp->xsk_info->ring_stats.tx_npkts += sp->pckt_count;
+            sp->xsk_info->outstanding_tx += sp->pckt_count;
+            while(sp->xsk_info->outstanding_tx != 0){
```

Comment:
this line is responsible for "waiting" until all the packets are sent in a given batch

Reply:
Not threadsafe however without any threading whatsoever its fine.